TempleEight/docs/guide/access-control.md

```diff
@@ -3,3 +3,136 @@ id: access-control
+  name: string;
+  #auth;
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+Since the project has authentication, every service actually has implicit default access control tags defined:
+
+```templefile
+ExampleUser: service {
+  name: string;
+  #auth;
+  // implicit #readable(by: this);
+  // implicit #writable(by: this);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  // implicit #readable(by: this);
+  // implicit #writable(by: this);
+}
+```
+
+The `by: this` means that only the authenticated user that created an entity in a service can be the one to read (make `READ` or `LIST` requests) the entity or write (make `UPDATE` or `DELETE` requests) to the entity.
+
+For example, if User A makes a `CREATE` request to the `ExampleService` service with their access token, the resulting entity created can only be read or written by User A.
+If User B then tries to make an `UPDATE` request to User A's entity with their own access token, User B will receive a `401 Unauthorized` error.
+
+If `by: all` is used, any user can make the corresponding requests unimpeded. So if instead we define `ExampleService` like this:
+
+```templefile
+ExampleService: service {
+  foo: string;
+  bar: int;
+  #readable(by: all);
+  #writable(by: all);
+}
+```
+
+When User A makes a `CREATE` request to this `ExampleService`, the resulting entity can be read and written by any other authenticated user. Good news for User B.
+
+The `#readable` and `#writable` tags can be set to differing values, for example:
+
+```templefile
+ExampleService: service {
+  foo: string;
+  bar: int;
+  #readable(by: all);
+  #writable(by: this);
+}
+```
+
+This means that any authenticated user can `READ` or `LIST` (if defined) a created `ExampleService` entity, but only the user that created the entity may `UPDATE` or `DELETE` it.
+
+### Defaults
+For projects with authentication, `#readable` and `#writable` behaviour defaults to `by: this` for security, forcing `by: all` to specified if desired.
+This reduces the likelihood of entities being exposed to users that should not have access.
+
+## Projects without Authentication
+For projects without authentication it doesn't really make sense to have access control, since users cannot be identified.
+Therefore all services have `#readable(by: all)` and `#writable(by: all)` implicitly defined on them:
+
+```templefile
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  // implicit #readable(by: all);
+  // implicit #writable(by: all);
+}
+```
+
+## Limitations
+We acknowledge that this current implementation is limited in the granularity of access control it can support.
+Access roles is top of the list of features we wish to support next.
+
+In the meantime, for projects with auth, the authentication token of a request is exposed to the hooks, so you can add your own access control there.
```

Comment:
Hooks can only restrict access, not widen access compared to what is specified here.

Reply:
I don't think what's written here says that, but I can see how it could be misunderstood. I'll clear it up